src/aeso_builtins.erl

```diff
@@ -123,11 +153,48 @@ check_event_type(EvtName, Type, Icode) ->
         false -> error({EvtName, payload_should_be_string, is, VMType})
     end.
 
+bfun(B, F, Args) ->
+    {IArgs, IExpr, IRet} = erlang:apply(?MODULE, F, Args),
+    {{builtin, B}, [private], IArgs, IExpr, IRet}.
```

Comment:
Can avoid `erlang:apply` and having to export the `builtin_` functions by moving the call to `builtin_function`:
```erlang
bfun(B, {IArgs, IExpr, IRet}) ->
    {{builtin, B}, [private], IArgs, IExpr, IRet}.

builtin_function(BF) ->
    case BF of
        {event, EventT} -> bfun(BF, builtin_event(EventT));
        ...
```